lnwallet/chanfunding/wallet_assembler.go

```diff
@@ -305,10 +352,41 @@ func (w *WalletAssembler) ProvisionChannel(r *Request) (Intent, error) {
 		// we will call the specialized coin selection function for
 		// that.
 		case r.FundUpToMaxAmt != 0 && r.MinFundAmt != 0:
+
+			// We need to ensure, that manullay selected coins,
+			// which are spent entirely on the channel funding,
+			// leave enough funds in the wallet to cover for a
+			// reserve.
+			reserve := r.WalletReserve
+			if len(manuallySelectedCoins) > 0 {
+				sumCoins := func(coins []Coin) btcutil.Amount {
+					var sum btcutil.Amount
+					for _, coin := range coins {
+						sum += btcutil.Amount(
+							coin.Value,
+						)
+					}
+
+					return sum
+				}
+
+				sumManual := sumCoins(manuallySelectedCoins)
+				sumAll := sumCoins(allCoins)
+
+				if sumAll-sumManual >= reserve {
+					// If sufficient reserve funds are
+					// available we don't have to provide
+					// for it during coin selection. The
+					// manually selected coins can be spent
+					// entirely on the channel funding.
+					reserve = 0
+				}
+			}
+
```

Comment:
So I think we need to decide if we _can_ (from the PoV of the external API contract) add more coins, or if we should fail here if the user has specified custom coins.